### PR TITLE
Jwt improvement part1

### PR DESCRIPTION
**Pull request by hardikmashru**
## 🔹 Jira Ticket(s)

* [MOB-7981](https://iterable.atlassian.net/browse/MOB-7981)

## ✏️ Description

> Please provide a brief description of what this pull request does.

[MOB-7981]: https://iterable.atlassian.net/browse/MOB-7981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: swift-sdk/Internal/RequestProcessorUtil.swift [hunk 1]
```diff
@@ -112,4 +117,16 @@ struct RequestProcessorUtil {
             ITBError(toLog)
         }
     }
+    
+    private static func resetAuthRetries(authManager: IterableAuthManagerProtocol?, requestIdentifier: String) {
+        if requestIdentifier != "disableDevice" {
```
Comment (Ayyanchira) [inline swift-sdk/Internal/RequestProcessorUtil.swift:122]: Use of constants will be great here

File: swift-sdk/IterableAuthManagerProtocol.swift [hunk 2]
```diff
@@ -7,7 +7,11 @@ import Foundation
...
     func getAuthToken() -> String?
     func resetFailedAuthCount()
-    func requestNewAuthToken(hasFailedPriorAuth: Bool, onSuccess: ((String?) -> Void)?)
+    func requestNewAuthToken(hasFailedPriorAuth: Bool, onSuccess: ((String?) -> Void)?, shouldIgnoreRetryPolicy: Bool)
+    func scheduleAuthTokenRefreshTimer(interval: TimeInterval, isScheduledRefresh: Bool, successCallback: AuthTokenRetrievalHandler?)
     func setNewToken(_ newToken: String)
     func logoutUser()
+    func pauseAuthRetries(_ pauseAuthRetry: Bool)
+    func setIsLastAuthTokenValid(_ isValid: Bool)
+    func getNextRetryInterval() -> Double
```
Comment (Ayyanchira) [inline swift-sdk/IterableAuthManagerProtocol.swift:16]: Why is in - `JWT-InSettings` - branch
`   func getRetryCount() -> Int `
  `  func getPauseAuthRetry() -> Bool`
were added? 

It looks like adding these two functionalities will deviate the SDK to have different source of truth for total retry count and getPauseAuthRetry. 
We should limit the protocol methods with whats here in this PR

**Review by Ayyanchira — COMMENTED** [review]
Added some comments. 

Also - when I run this branch with a separate dev app, it does not seem to respect the retry policy set. No retries are made

**Comment by Ayyanchira** [comment]
Can the base branch be changed from master to a feature branch just like how it was done for Android counterpart? @hardikmashru ?

File: swift-sdk/Internal/AuthManager.swift [hunk 3]
```diff
@@ -35,26 +37,44 @@ class AuthManager: IterableAuthManagerProtocol {
...
     
-    func requestNewAuthToken(hasFailedPriorAuth: Bool = false, onSuccess: AuthTokenRetrievalHandler? = nil) {
+    func requestNewAuthToken(hasFailedPriorAuth: Bool = false,
+                             onSuccess: AuthTokenRetrievalHandler? = nil,
+                             shouldIgnoreRetryPolicy: Bool) {
         ITBInfo()
         
-        guard !pendingAuth else {
-            return
-        }
```
[the remark this answers is not in the document]
Comment (hardikmashru) [reply swift-sdk/Internal/AuthManager.swift:43]: yes and we have combined it with other conditions with OR

File: swift-sdk/Internal/RequestProcessorUtil.swift (hunk 1 above)
Comment (hardikmashru) [reply swift-sdk/Internal/RequestProcessorUtil.swift:122]: Done

File: swift-sdk/IterableAuthManagerProtocol.swift (hunk 2 above)
Comment (hardikmashru) [reply swift-sdk/IterableAuthManagerProtocol.swift:16]: Yes @Ayyanchira actually to get the JWT demo working properly (with correct values of retry count and to pause it using authmanager) we had to expose authManager along with these additional methods.

**Comment by hardikmashru** [comment]
> Can the base branch be changed from master to a feature branch just like how it was done for Android counterpart? @hardikmashru ?

done

**Comment by Ayyanchira** [comment]
Merging to Feature branch to proceed to check Part 2 with Part 1

**Pull request merged**